lightning/src/ln/msgs.rs

```diff
@@ -1998,15 +2023,17 @@ mod tests {
 	#[test]
 	fn encoding_final_onion_hop_data() {
 		let mut msg = msgs::OnionHopData {
-			format: OnionHopDataFormat::FinalNode,
+			format: OnionHopDataFormat::FinalNode {
+				payment_data: None,
```

Comment:
Add/augment a test where payment_data is non-`None`?

Reply:
Good call. Turned up an error where the total_msat is a varint, not a regular u64.